src/raw_vec.rs

```diff
@@ -537,7 +535,7 @@ where
 
     let memory = if let Some((ptr, old_layout)) = current_memory {
         debug_assert_eq!(old_layout.align(), new_layout.align());
-        unsafe { alloc.grow(ptr, old_layout, new_layout.size()) }
+        unsafe { alloc.grow(ptr, old_layout, new_layout) }
```

Comment:
Upstream, the following change gave a small performance boost in debug code:
```suggestion
        unsafe {
            // The allocator checks for alignment equality
            core::intrinsics::assume(old_layout.align() == new_layout.align());
            alloc.grow(ptr, old_layout, new_layout)
        }
```